lib/components/component.dart

```diff
@@ -34,38 +34,41 @@ abstract class Component {
   /// Use [Resizable] to save the gameSize in a component.
   void onGameResize(Vector2 gameSize) {}
 
+  /// Whether this component should be removed or not.
+  ///
+  /// It will be checked once per component per tick, and if it is true, [BaseGame] will remove it.
+  bool shouldRemove = false;
+
+  /// Remove the component from the game it is added to in the next tick
+  void remove() => shouldRemove = true;
+
   /// Whether this component has been loaded yet. If not loaded, [BaseGame] will not try to render it.
   ///
   /// Sprite based components can use this to let [BaseGame] know not to try to render when the [Sprite] has not been loaded yet.
   /// Note that for a more consistent experience, you can pre-load all your assets beforehand with Flame.images.loadAll.
-  bool loaded() => true;
-
-  /// Whether this should be destroyed or not.
-  ///
-  /// It will be called once per component per loop, and if it returns true, [BaseGame] will mark your component for deletion and remove it before the next loop.
-  bool destroy() => false;
+  bool loaded = true;
 
   /// Whether this component is HUD object or not.
   ///
   /// HUD objects ignore the [BaseGame.camera] when rendered (so their position coordinates are considered relative to the device screen).
-  bool isHud() => false;
+  bool isHud = false;
 
   /// Render priority of this component. This allows you to control the order in which your components are rendered.
   ///
   /// Components are always updated and rendered in the order defined by this number.
   /// The smaller the priority, the sooner your component will be updated/rendered.
   /// It can be any integer (negative, zero, or positive).
   /// If two components share the same priority, they will probably be drawn in the order they were added.
-  int priority() => 0;
+  int priority = 0;
```

Comment:
We need to think about how to handle the changing of this.

Changing this variable will not change the render order. That is why this was a method.

Honestly I don't have any solution for this but we need to think on something hahahaha 😅

Reply:
Isn't this doing that? :)
https://github.com/flame-engine/flame/pull/527/files#diff-39ef6a651074ea090af849d06cafd156e3e49ddc3bcde638bb83b9d3a8d7069eR26

Reply:
As far as I rememeber, not really, that only executes when the element is added

Reply:
Ah, it doesn't resort it later?

Reply:
Nope :(

That is why we have this card on the board actually: https://github.com/flame-engine/flame/projects/1#card-40463639

Reply:
Nope :(

That is why we have this card on the board actually: https://github.com/flame-engine/flame/projects/1#card-40463639

Reply:
I just tried, it is the same behavior having it as a method as having it as a field, both of them only puts them in the correct position on insertion.